Northwind.Console/Configuration/ConsoleServicesConfiguration.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Net.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Northwind.Console.Interfaces.Services;
+using Northwind.Console.Options;
+using Northwind.Console.Services;
+
+namespace Northwind.Web.Configuration
+{
+    public static class ConsoleServicesConfiguration
+    {
+        public static IServiceCollection AddConsoleServices(this IServiceCollection services)
+        {
+            return services
+                .AddServices()
+                .AddHttpClient();
+        }
+
+        private static IServiceCollection AddServices(this IServiceCollection services)
+        {
+            services.AddScoped<INorthwindApiService, NorthwindApiService>();
+
+            return services;
+        }
+
+        private static IServiceCollection AddHttpClient(this IServiceCollection services)
+        {
+            services.AddSingleton<HttpClient>(services => new HttpClient

```

Comment:
HttpClient as a singleton can lead to changing DNS problem. Can you please use IHttpClinetFactory here, wich is added since .net core 2.1.
IHttpClientFactory can be registered by calling `AddHttpClient`:
```
services.AddHttpClient("myapiclient", c =>
{
    c.BaseAddress = new Uri("https://myapi.com/");    
});
```
Here is more about `IHttpClientFactory` and the problems it solves: https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests
Here is about usage patterns and examples: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-5.0#consumption-patterns

Reply:
Thank you! I've fixed it